modules/apps/asset/asset-categories-navigation-web/src/main/java/com/liferay/asset/categories/navigation/web/internal/portlet/action/AssetCategoriesNavigationConfigurationAction.java

```diff
@@ -29,4 +49,118 @@ public String getJspPath(HttpServletRequest httpServletRequest) {
 		return "/configuration.jsp";
 	}
 
+	@Override
+	protected void postProcess(
+			long companyId, PortletRequest portletRequest,
+			PortletPreferences portletPreferences)
+		throws PortalException {
+
+		super.postProcess(companyId, portletRequest, portletPreferences);
+
+		boolean allAssetVocabularies = GetterUtil.getBoolean(
+			portletPreferences.getValue("allAssetVocabularies", null));
+
+		if (allAssetVocabularies) {
+			return;
+		}
+
+		String assetVocabularyIdsString = portletPreferences.getValue(
+			"assetVocabularyIds", null);
+
+		if (Validator.isNull(assetVocabularyIdsString)) {
+			return;
+		}
+
+		long[] assetVocabularyIds = GetterUtil.getLongValues(
+			StringUtil.split(assetVocabularyIdsString, ','));
+
+		Map<Long, List<String>> groupAssetVocabularyIdsMap =
+			_getGroupAssetVocabularyIdsMap(assetVocabularyIds);
+
+		try {
+			_resetPortletPreferences(portletPreferences);
+
+			_setPortletPreferences(
+				portletPreferences, groupAssetVocabularyIdsMap);
+
+			portletPreferences.reset("assetVocabularyIds");
+		}
+		catch (ReadOnlyException readOnlyException) {
+			throw new SystemException(readOnlyException);
+		}
+	}
+
+	private Map<Long, List<String>> _getGroupAssetVocabularyIdsMap(
+			long[] assetVocabularyIds)
+		throws PortalException {
+
+		Map<Long, List<String>> groupAssetVocabularyIdsMap = new HashMap<>();
+
+		for (long assetVocabularyId : assetVocabularyIds) {
+			AssetVocabulary assetVocabulary =
+				_assetVocabularyService.fetchVocabulary(assetVocabularyId);
+
+			if (assetVocabulary == null) {
+				continue;
+			}
+
+			List<String> groupAssetVocabularyExternalReferenceCodes =
+				groupAssetVocabularyIdsMap.computeIfAbsent(
+					assetVocabulary.getGroupId(), key -> new ArrayList<>());
+
+			groupAssetVocabularyExternalReferenceCodes.add(
+				assetVocabulary.getExternalReferenceCode());
+		}
+
+		return groupAssetVocabularyIdsMap;
+	}
```

Comment:
Shouldn't it be named `_getGroupAssetVocabularyExternalReferenceCodesMap`?

Reply:
Yes, my bad, thanks @adolfopa!